fix: correct skill file API names, add workflow guidance, fix publish tx (#1832)
tari-project/tari-dan: docs/skills/windsurf/SKILL.md

    .find_map(|s| s.substate_id.as_resource_address().filter(|a| *a != XTR))
    .expect("resource address in receipt");

// Find a template address (returns PublishedTemplateAddress)
let template_addr = receipt.diff_summary.upped
    .iter()
    .find_map(|s| s.substate_id.as_template())
    .expect("template address in receipt");
```

> > **Note:** `as_component_address()` and `as_resource_address()` return `ComponentAddress` and `ResourceAddress` directly. For templates, `as_template()` returns `PublishedTemplateAddress`. Call `.as_template_address()` on the result if you need the underlying `TemplateAddress` (a `Hash32`).

Reviewer: ```suggestion
> **Note:** Call `as_component_address()`,  `as_resource_address()` and `as_template()` on `SubstateId` to return `ComponentAddress`, `ResourceAddress` and `PublishedTemplateAddress` respectively. Call `.as_template_address()` on `PublishedTemplateAddress` for the underlying `TemplateAddress` (alias for `Hash32`).
```
